src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -91,6 +91,45 @@
 )
 
 
+@postgres_dialect.segment()
+class TimeZoneGrammar(BaseSegment):
+    """Literal Date Time with optional casting to Time Zone."""
+
+    type = "time_zone_grammar"
+    match_grammar = AnyNumberOf(
```

Comment:
I'm not a frequent Postgres user, but I'm surprised this is `AnyNumberOf()`!

Reply:
Agree but it is - see comment at the bottom of this https://github.com/sqlfluff/sqlfluff/pull/1398#issuecomment-918991891

SQL dialects are weird 🤷‍♂️